fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AuthorizerStmtVisitor.java

```diff
@@ -338,6 +484,67 @@ void checkSelectTableAction(ConnectContext context, Map<TableName, Relation> all
         }
     }
 
+    private void checkCanSelectFromColumns(ConnectContext context, Map<TableName, Set<String>> allTouchedTableColumns,
+                                           Map<TableName, Relation> allTouchedTables) {
+        HashSet<TableName> usedTables = new HashSet<>(allTouchedTables.keySet());
+        HashSet<TableName> usedColOfTables = new HashSet<>(allTouchedTableColumns.keySet());
+        usedTables.removeAll(usedColOfTables);
+        if (!usedTables.isEmpty()) {
+            String warnMsg = String.format("The usage columns information of some actually used tables " +
+                    "has not been successfully collected. tables: %s", usedTables);
+            LOG.warn(warnMsg);
+            // used `*` to fill these tables column privileges
+            for (TableName usedTable : usedTables) {
+                try {
+                    Authorizer.checkColumnsAction(context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
+                            usedTable, Collections.singleton("*"), PrivilegeType.SELECT);
+                } catch (AccessDeniedException e) {
+                    AccessDeniedException.reportAccessDenied(
+                            usedTable.getCatalog(),
+                            context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
+                            PrivilegeType.SELECT.name(), ObjectType.COLUMN.name(),
+                            String.join(".", usedTable.getTbl(), "*"));
+                }
+            }
+        }
+        for (Map.Entry<TableName, Set<String>> tableColumns : allTouchedTableColumns.entrySet()) {
+            TableName tableName = tableColumns.getKey();
+            Set<String> columns = tableColumns.getValue();
+            Relation relation = allTouchedTables.get(tableName);
+            if (relation == null) {
+                String warnMsg = String.format("Some used columns of tables were incorrectly collected. table: %s",
+                        tableName);
+                LOG.warn(warnMsg);
```

Comment:
I think we should throw an exception directly after this logging, the root cause of the two situations is the same, i.e. some kind of implementation bug. But it's also different, because for the above situation we actually check some kind of `BIGGER` privilege, it's safe. But for this situation, we just ignore the privilege check, it's unsafe.

Reply:
You are right, shall we throw AccessDeniedException or some other type Exception?

Reply:
AccessDeniedException is fine